table_to_json.py

```diff
@@ -4,82 +4,97 @@
 from bs4 import BeautifulSoup
 
 
-'''
-Converts table from html document to json format
-'''
+class TableToJSON:
+    '''
+    Converts table from html document to json format
+    '''
 
-#return matrix maked from table for specified tag.
-def create_matrix(tag_type):
-    temporary_table = np.empty(shape=(len(converted_tr), table_width), dtype=object)
-    for idx, single_tr in enumerate(converted_tr):
-        column_shift = 0  # when encouter colspan add number
-        for idxtd, single_td in enumerate(single_tr.find_all(lambda tag: tag.name == 'td' or tag.name == 'th')):
-            if single_td.name == tag_type:
-                while temporary_table[idx, idxtd + column_shift] is not None:
-                    column_shift += 1
+    def __init__(self, file_name):
+        with open(file_name) as fp:
+            soup = BeautifulSoup(fp.read(), "lxml")
+        self.converted_tr = soup.find_all('tr')
```

Comment:
wszystko co jest zależne od soup wrzuciłbym do innej metody i wywołałbym to tu z wcięciem (with open jest blokiem o podobnym zakresie co if, while itp.). Wszystko co zależne od otwarcia pliku powinno być tu z wcięciem, ew. sprawdzane czy plik rzeczywiscie udało się odczytać.